types/eslint/index.d.ts

```diff
@@ -1085,9 +1086,41 @@ export namespace Linter {
     /**
      * Parser options.
      *
-     * @see [Specifying Parser Options](https://eslint.org/docs/latest/use/configure/language-options-deprecated#specifying-parser-options)
+     * This properties are used to pass options directly to parsers. These options are always parser-specific,
+     * so you’ll need to check the documentation of the parser you’re using for available options.
+     *
+     * @see [Specifying Parser Options](https://eslint.org/docs/latest/use/configure/language-options#specifying-parser-options)
+     * @see [Configure Parser Options](https://eslint.org/docs/latest/use/configure/parser#configure-parser-options)
      */
     interface ParserOptions {
```

Comment:
memo: The `ParserOptions` type for Legacy Config has been renamed (`ParserOptions` => `LegacyParserOptions`). This may be breaking changes.

Reply:
Perhaps we should use the type name for legacy config to `ParserOptions` and the type name for flat config to `FlatParserOptions`. That way, the type name for legacy config will be preserved and no breaking changes will occur.

Reply:
resolved: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/70385/commits/e195e53b63f77fcb0ce9b18a566d67606c4e19ea